meteor/sofie-server/data-access/repositories/mongo/mongo-part-repository.ts

```diff
@@ -33,4 +34,22 @@ export class MongoPartRepository extends BaseMongoRepository implements PartRepo
 			})
 		)
 	}
+
+	public async deleteParts(segmentId: string): Promise<void> {
+		this.assertDatabaseConnection(this.deleteParts.name)
+		const parts = await this.getParts(segmentId)
+
+		parts.map((part) => part.id).forEach(async (id) => await this.pieceRepository.deletePieces(id))
```

Comment:
I know that the `id` is taken from each of the parts, but i am not sure if i like it doing so implicitly, instead of current explicit way.

Say we were to rename the property, would it also get renamed in your suggested way?
My gut feeling says no, but i don't know if that is true.

Unless others support your suggested way, i will not be changing this - atleast for now.

Reply:
> parts.forEach(({ id })

I am personally in favour for the `map` instead of the above suggestions. 
The suggestion is a bit too implicit.

But might as well just keep it simple and do: `parts.forEach(part => this.pieceRepository.deletePieces(part.id))`

Reply:
As suggested by Lindved i have cut away the map.